src/odemis/driver/simulated.py

```diff
@@ -339,3 +238,197 @@ def __init__(self, name, role, positions, has_pressure=False, **kwargs):
+            "unit" (str): optional, the unit of the VA, defaults to ""
+            "range" (float, float): optional, min/max of the VA, defaults to None
+            "choices" (dict): optional, possible values available to the VA, defaults to None, will be ignored if range
+                              is defined
+        axes (dict (string -> dict (string -> any))): dict mapping desired axis names to dicts with axis properties. The
+        axis property dict can contain the following keys:
+            "unit" (str): optional, unit of the axis, defaults to "m"
+            "range" (float, float): optional, min/max of the axis, defaults to (-0.1, 0.1)
+            "choices" (dict): optional, alternative to ranges, these are the choices of the axis
+            "speed" (float, float): optional, allowable range of speeds, defaults to (0., 10.)
+
+        """
+        # Create desired VA's
+        if vas:
+            for vaname, vaprop in vas.items():
+                # Guess an appropriate VA type based on the initial value and the presence of range or choices
+                vavaluetype = vaprop["value"].__class__.__name__.capitalize()
+                if vavaluetype == "Bool":
+                    vavaluetype = "Boolean"
+                elif vavaluetype == "Str":
+                    vavaluetype = "String"
+                if "range" in vaprop:
+                    varangetype = "Continuous"
+                elif "choices" in vaprop:
+                    varangetype = "Enumerated"
+                else:
+                    varangetype = "VA"
+                vatype = vavaluetype + varangetype
+
+                if "value" not in vaprop:
+                    vaprop["value"] = 0.0  # default value
+
+                try:
+                    vaclass = getattr(model, vatype)
+                except KeyError:
+                    raise KeyError(f"VA type {vatype}, guessed for VA {vaname}, does not exist.")
+                va = vaclass(**vaprop)
+                setattr(self, vaname, va)
+
+        # Create desired axes
+        axes_def = {}
+        if axes:
+            self._position = {}
+            init_speed = {}
+            for axisname, axisprop in axes.items():
+                init_speed[axisname] = 1.0  # we are fast!
+                if "range" not in axisprop and "choices" not in axisprop:  # if no range nor choices are defined
+                    axisprop["range"] = (-0.1, 0.1)  # use the default range
+                if "speed" not in axisprop:
+                    axisprop["speed"] = (0., 10.)  # default speed
+                axes_def[axisname] = model.Axis(**axisprop)
+                if "range" in axisprop:
+                    self._position[axisname] = (axisprop["range"][0] + axisprop["range"][1]) / 2  # start at the centre
+                else:
+                    self._position[axisname] = list(axisprop["choices"].values())[0]  # start at the first value
```

Comment:
The axes support two (!) kinds of choices: set and dict. In case of dict, it's position -> user-friendly name. So at least you should use .keys().
However, a more fancy way, which would also work in case of a set would be to write `next(iter(axisprop["choices"]))`. As a comment, to be more precise indicate "pick one value arbitrarily" (as set and dict are not ordered).